lib/logger/lib/logger.ex

```diff
@@ -592,4 +600,28 @@ defmodule Logger do
 
   defp notify(:sync, msg),  do: GenEvent.sync_notify(Logger, msg)
   defp notify(:async, msg), do: GenEvent.notify(Logger, msg)
+
+  defp handle_unused_variable_warnings(data, caller) do
+    # We collect all the ASTs of variables (leaving `data` unchanged); we only
+    # want the variables that figure in `caller.vars`, as the AST for calls to
+    # local 0-arity functions without parens is the same as the AST for
+    # variables.
+    {^data, logged_vars} = Macro.postwalk(data, [], fn
+      {name, _meta, scope} = var, acc when is_atom(name) and is_atom(scope) ->
+        if {name, scope} in caller.vars, do: {var, [var | acc]}, else: {var, acc}
+      ast, acc ->
+        {ast, acc}
+    end)
+
+    assignments =
+      logged_vars
+      |> Enum.reverse()
+      |> Enum.uniq_by(fn {name, _, _} -> name end)
```

Comment:
It needs to be uniq by "name and scope" OR we should only collect vars with scope == nil above. I think we don't warn if scope != nil, so we can simply get the names inside postwalk, ignore the scope, and simply call `uniq` here.